x-pack/plugins/ml/server/models/data_frame_service/data_frame.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import _ from 'lodash';
+
+export type callWithRequestType = (action: string, params?: any) => Promise<any>;
```

Comment:
i'm unsure about having these endpoint errors wrapped in `Boom.badRequest`
The real error code from the server will be lost.
`/x-pack/plugins/ml/server/routes/anomaly_detectors.js` is similar to this and just makes the `callWithRequest` call inside the routing file.
I guess what i'm getting at is, this file might not be needed it all it's doing is wrapping the endpoint calls with try/catches

Reply:
thanks for pointing that out! i was able to get rid of that whole file in 1334091. PR is now under 3000 lines again 😄